watsong/spotify.py

```diff
@@ -3,20 +3,60 @@
 """
 import heapq
 import pickle
-from typing import Any, Dict, Iterator, List, Optional, TypeVar
+from typing import Any, Dict, Generic, List, Optional, cast
 
+import flask
 import spotipy
 from spotipy.oauth2 import SpotifyOAuth
+from typing_extensions import TypedDict
 
-from .structures import Album, AlbumDescription, Feel, Song
 from . import util
+from .structures import Album, AlbumDescription, Feel, Song
+from .test.spotify_mocks import between_worlds_mock
 
 # These are also stored in the environment but it's easier to leave them here
 # since it causes some problems in how I run it if I use the environment variables
 CLIENT_ID = "8170c7110cfb4503af349a6a8ea22fd3"
 CLIENT_SECRET = "0be6c71210bd495ab3f75e9b7f8a8935"
 USERNAME = "rp5ukikcsq2vjzakx29pxazlq"
 
+# region types
```

Comment:
I was struggling to understand the return format of the spotify API and I was running into cache issues, so I added some static types so that I could understand the data structures. We can get rid of them if we think it's too clunky.